### ISSUE TITLE
Allow users to edit or fill in data dictionary

### ISSUE THREAD

**Issue by surchs** [neurobagel/legacy_annotation_tool#185]
As a user, while I am annotating data, I want to be able to fill in or edit the "Description" section of columns and values, so that I can create or update a data dictionary for this dataset while I am creating the annotations.

Example: I have a dataset that only has a `participants.tsv` file and no data dictionary. Thus, on the categorization page the description section for each column is empty. I now want to add a description for some of the columns. To do that I click into the (currently empty) description field for the column and start typing text. My expectation is that:

- I can type in the description section of the columns, even if this section is empty
- the text I have entered is stored without any further action (i.e. what is currently visible is what is also stored in the system)
- there is an option to download the data dictionary as it is currently stored in the system, and this process will export my manual edits to me.

**Comment by jarmoza** [comment]
Since there is a view of the data dictionary when it is loaded on the index page, there may be an expectation that the data dictionary is also editable there. The same would go for the participants.tsv. We should consider consistency in the UI. It may make the most sense to have this done on the index page and change the view of the data dictionary to an editable table – and gray out the participants tsv (also displayed in table form). There should be a max row view as well, of course, so that both data table and dictionary remain visible at all times.

**Comment by jarmoza** [comment]
@surchs How do you feel about this feature? It could be a relatively simple change now that we are moving to the tool's output being an annotated data dictionary.

**Comment by surchs** [comment]
Yeah, I think it will be much easier to implement. We could flesh out the idea a little more and then add it to the next milestone. wdyt?

**Comment by jarmoza** [comment]
@surchs Likewise, do you have any further thoughts on this feature/when to implement? Coincidentally, we technically have the ability to do this now by reverting the `textarea` for the data dictionary on the landing page back to using a `v-model` attribute. Definitely not as slick as the suggested above UI, but users can hand edit the data dictionary in the text box and it the edits should be reflected in the current ...`userProvided` data dictionary.

**Comment by surchs** [comment]
> do you have any further thoughts on this feature/when to implement? 

I think we can sneak it into one of the next releases where we have some extra capacity with mid-low priority. At the moment, nothing critical depends on this (this may change)

> users can hand edit the data dictionary in the text box and it the edits should be reflected 

I see. That's an interesting solution, but I would vote for doing this cleanly with a nice user experience / UI directly.

**Comment by jarmoza** [comment]
> I see. That's an interesting solution, but I would vote for doing this cleanly with a nice user experience / UI directly.

Fully agreed. Just a funny coincidence.

**Comment by Manya-tech** [comment]
Hello

I am Manya Gupta. I was using the annotation tool as a potential GSoC contributor and struggled with this problem. Then I read the documentation and there was no mention of how to add description or that this feature is not available yet. Looking at the images provided in the guide made me think that maybe I missed the part mentioning how to add descriptions leading me to read the it a few more times. I would like to suggest updating the documentation to enhance the user experience by mentioning that this feature will be added in the next release. 

Hope this suggestion is helpful!

**Comment by rmanaem** [comment]
Hi @Manya-tech 
Thanks for the feedback! Would you like to open an issue for us to keep track of?
